gen/modules/general.h

```diff
@@ -279,3 +281,94 @@ struct user_desc {
 #define BLKSSZGET  0x1268
 #define BLKPBSZGET 0x127B
 #endif
+
+
+/* -- from socket.h -- */
```

Comment:
Did you copy this from glibc or musl? If from glibc, that would mean this crate would need to be lgpl licensed with this PR. If from musl you will need to explicitly say that the apache-2.0 license doesn't apply to this and add the musl license header to this part of the file.

Reply:
The thing to do here is to only include what's necessary to make compatible bindings. Specifically, exclude all comments from the original code (we can add our own comments if we want, but it's not necessary), and exclude things like the inline functions and function-style macros, which bindgen can't bind to anyway.

Reply:
Atm just trying to get things to compile 😓
Ideally we would get all this stuff directly out of `socket.h`, I don't quite understand why it does not, any ideas? 

I will need the macros as well, they are nicely done in libc.rs, I could add them to either here, or directly in `rustix` any preferences where those should live? 

Also libc.rs has the benefit of being MIT/Apache2 so could simply skip the c impl part and just base the rust version of that.

Reply:
As I understand these structs are part of the libc headers and not the linux headers.

Reply:
@bjorn3 I guess my question boils down to given these live in `sys/socket.h` (libc land) where should I put implementations of this. Is this the right place, or should they live directly in `rustix/imp/linux_raw`, or is there another place?

Reply:
The structs can be defined here in general.h. They should match the Linux ones in linux's socket.h (which are unavailable to userspace because they appear to be erroneously guarded by `ifdef __KERNEL__`). Note that kernel structs don't always match libc structs, though they're often similar.

The `CMSG_*` functions and function-style macros will need to be written in Rust, as bindgen can't generate bindings for them. This code could either live in linux-raw-sys in src/lib.rs or in rustix under src/imp/linux_raw. It seems slightly nicer to put it in src/imp/linux_raw, so keep linux-raw-sys simple.

Reply:
updated to minimize the additions here in `general.h`